### PR TITLE
Refactor Brave Ads in preparation for merging Publisher Ads

### PR DESCRIPTION
**Pull request by tmancey**
Resolves https://github.com/brave/brave-browser/issues/8294

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

- Confirm ad notifications are shown and the user is rewarded for viewing an ad

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by tmancey** [comment]
Cleaned up data structures, Renamed instances of `Ad` to `AdNotification` ready for `PublisherAds`. Renamed `uuid` to `creative_instance_id` then renamed `id` to `uuid` to match catalog/server-side. Removed `const` where not needed. Added `const` where needed.

**Comment by NejcZdovc** [comment]
@tmancey CI is failing on build. https://ci.brave.com/job/brave-browser-build-pr/job/issues%252F8294/2/execution/node/481/log/

**Comment by tmancey** [comment]
> @tmancey CI is failing on build. https://ci.brave.com/job/brave-browser-build-pr/job/issues%252F8294/2/execution/node/481/log/

Fixed, thanks

**Comment by tmancey** [comment]
Unrelated CI TestSuite Security failure

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc [hunk 1]
```diff
@@ -1199,86 +1199,89 @@ void LedgerImpl::SetCatalogIssuers(const std::string& info) {
...
-  notification_info->text = notification_info_ads.text;
-  notification_info->url = notification_info_ads.url;
-  notification_info->uuid = notification_info_ads.uuid;
-
-  switch (notification_info_ads.type.value()) {
-    case ads::ConfirmationType::UNKNOWN: {
-      notification_info->type = confirmations::ConfirmationType::UNKNOWN;
+  notification_info->title = notification_info_ads.title;
+  notification_info->body = notification_info_ads.body;
+  notification_info->target_url = notification_info_ads.target_url;
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc]: Suggest: AdNotificationInfo constructor to consume notification_info_ads & creative_instance

File: components/brave_ads/browser/bundle_state_database.cc [hunk 2]
```diff
@@ -398,54 +402,67 @@ bool BundleStateDatabase::InsertOrUpdateAdInfoCategory(
...
+bool BundleStateDatabase::GetCreativeAdNotifications(
+    const std::vector<std::string>& categories,
+    ads::CreativeAdNotificationList* ads) {
   DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
 
+  DCHECK(ads);
+  if (!ads) {
+    return false;
+  }
+
```
Comment (masparrow) [inline components/brave_ads/browser/bundle_state_database.cc]: nit: as discussed, Chromium standards say do not DCHECK and then guard... so remove the if? Even if I still don't agree with it... 😄

File: vendor/brave-ios/Ads/BATAdsNotification.mm [hunk 3]
```diff
@@ -3,34 +3,34 @@
...
 @end
 
 @implementation BATAdsNotification
 
-- (instancetype)initWithNotificationInfo:(const ads::NotificationInfo&)info
+- (instancetype)initWithNotificationInfo:(const ads::AdNotificationInfo&)info
 {
   if ((self = [super init])) {
-    self.id = [NSString stringWithUTF8String:info.id.c_str()];
+    self.idterry = [NSString stringWithUTF8String:info.uuid.c_str()];
```
Comment (kylehickinson) [inline vendor/brave-ios/Ads/BATAdsNotification.mm]: `idterry`?

File: vendor/brave-ios/Ads/BATAdsNotification.mm [hunk 4]
```diff
@@ -3,34 +3,34 @@
...
+@property (nonatomic, copy) NSString *creativeInstanceId;
 @property (nonatomic, copy) NSString *creativeSetID;
 @property (nonatomic, copy) NSString *category;
-@property (nonatomic, copy) NSString *advertiser;
-@property (nonatomic, copy) NSString *text;
-@property (nonatomic, copy) NSURL *url;
-@property (nonatomic, copy) NSString *uuid;
+@property (nonatomic, copy) NSString *title;
+@property (nonatomic, copy) NSString *body;
+@property (nonatomic, copy) NSURL *target_url;
```
Comment (kylehickinson) [inline vendor/brave-ios/Ads/BATAdsNotification.mm]: Changes made here to `readwrite` overrides of the properties but not the public header (`BATAdsNotification.h`) properties

Comment (kylehickinson) [reply vendor/brave-ios/Ads/BATAdsNotification.mm]: Also wrong style for Obj-C `target_url` → `targetURL`

File: vendor/bat-native-ads/include/bat/ads/ads_history.h [hunk 5]
```diff
@@ -32,7 +32,7 @@ struct ADS_EXPORT AdsHistory {
     kDescendingOrder
   };
 
-  const std::string ToJson() const;
+  std::string ToJson() const;
```
[the remark this answers is not in the document]
Comment (tmancey) [reply vendor/bat-native-ads/include/bat/ads/ads_history.h]: in C++11 with its `rvalues` this convention doesn't make as much sense as the practice is obsolete

File: vendor/bat-native-ads/src/bat/ads/internal/error_helper.cc [hunk 6]
```diff
@@ -7,7 +7,7 @@
 
 namespace helper {
 
-const std::string GetDescription(const ads::Result result) {
+std::string GetDescription(const ads::Result result) {
```
[the remark this answers is not in the document]
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/error_helper.cc]: in C++11 with its `rvalues` this convention doesn't make as much sense as the practice is obsolete

File: vendor/brave-ios/Ads/BATAdsNotification.mm (hunk 3 above)
Comment (tmancey) [reply vendor/brave-ios/Ads/BATAdsNotification.mm]: Fixing now

Comment (tmancey) [reply vendor/brave-ios/Ads/BATAdsNotification.mm]: Fixed

File: vendor/brave-ios/Ads/BATAdsNotification.mm (hunk 4 above)
Comment (tmancey) [reply vendor/brave-ios/Ads/BATAdsNotification.mm]: Fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-ledger/src/bat/ledger/internal/ledger_impl.cc]: This PR is to mainly rename functions/variables ready for Publisher Ads, so will add to a future refactor ticket

File: vendor/bat-native-ads/src/bat/ads/internal/catalog_state.cc [hunk 7]
```diff
@@ -149,40 +142,39 @@ Result CatalogState::FromJson(
...
+          creative_info.type.code = code;
+          creative_info.type.name = type["name"].GetString();
+          creative_info.type.platform = type["platform"].GetString();
+          creative_info.type.version = type["version"].GetUint64();
+
+          // Payload
+          auto payload = creative["payload"].GetObject();
+          creative_info.payload.body = payload["body"].GetString();
+          creative_info.payload.title = payload["title"].GetString();
+          creative_info.payload.target_url = payload["targetUrl"].GetString();
```
[the remark this answers is not in the document]
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/catalog_state.cc]: This PR is to mainly rename functions/variables ready for Publisher Ads, so will add to a future refactor ticket

File: vendor/bat-native-ads/src/bat/ads/internal/bundle.cc [hunk 8]
```diff
@@ -121,35 +122,34 @@ std::unique_ptr<BundleState> Bundle::GenerateFromCatalog(
...
+        info.start_at_timestamp = campaign.start_at;
+        info.end_at_timestamp = campaign.end_at;
+        info.daily_cap = campaign.daily_cap;
+        info.advertiser_id = campaign.advertiser_id;
+        info.per_day = creative_set.per_day;
+        info.total_max = creative_set.total_max;
+        info.geo_targets = regions;
+        info.title = creative.payload.title;
+        info.body = creative.payload.body;
+        info.target_url = creative.payload.target_url;
```
[the remark this answers is not in the document]
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/bundle.cc]: This PR is to mainly rename functions/variables ready for Publisher Ads, so will add to a future refactor ticket

File: vendor/bat-native-ads/src/bat/ads/internal/error_helper.cc (hunk 6 above)
[the remark this answers is not in the document]
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/error_helper.cc]: in C++11 with its `rvalues` this convention doesn't make as much sense as the practice is obsolete

File: vendor/bat-native-ads/include/bat/ads/ads_history.h (hunk 5 above)
[the remark this answers is not in the document]
Comment (tmancey) [reply vendor/bat-native-ads/include/bat/ads/ads_history.h]: in C++11 with its `rvalues` this convention doesn't make as much sense as the practice is obsolete

File: components/brave_ads/browser/bundle_state_database.cc [hunk 9]
```diff
@@ -398,54 +402,67 @@ bool BundleStateDatabase::InsertOrUpdateAdInfoCategory(
...
           "INNER JOIN ad_info_category AS aic "
           "ON aic.ad_info_uuid = ai.uuid "
-          "WHERE aic.category_name = ? and "
+          "WHERE aic.category_name IN (?) and "
           "ai.start_timestamp <= strftime('%Y-%m-%d %H:%M', "
           "datetime('now','localtime')) and "
           "ai.end_timestamp >= strftime('%Y-%m-%d %H:%M', "
           "datetime('now','localtime'));"));
-  info_sql.BindString(0, category);
+  std::vector<std::string> quoted_categories;
```
Comment (tmancey) [inline components/brave_ads/browser/bundle_state_database.cc]: Business logic moved from `AdsServiceImpl` to fetch ads for multiple categories

**Comment by tmancey** [comment]
@NejcZdovc Other changes were to rename variables and functions to be inline with Publisher Ads

**Comment by NejcZdovc** [comment]
CI failed on android, everything else passed. Restarting android

**Pull request merged**